packages/apps/sale/src/components/molecules/CartLineActionCard/CartLineActionCard.tsx

```diff
@@ -76,66 +92,97 @@ const CartLineActionCard = ({
     }
   }, [handleTimeOut, diffQty]);
 
+  const handleConfirm = useCallback(() => {
+    fetchMatchingProduct({selectedVariants})
+      .then(res => {
+        const variantProduct = res?.data?.data?.[0];
+        if (variantProduct) {
+          dispatch(
+            (updateCartLine as any)({
+              cartLine,
+              qty: cartLine.qty,
+              variantProduct,
+              cartId,
+            }),
+          );
+        }
+      })
+      .catch(error => {
+        console.error('Error fetching matching product:', error);
```

Comment:
Use a toast here to show the error to the user. And maybe I missing something but it would be easier to use a slice function here, as the toast management will be automatic